cloud/services/compute/network.go

```diff
@@ -78,6 +87,22 @@ func (s *Service) DeleteNetwork() error {
 		return nil
 	}
 
+	// Delete Router.
+	router, err := s.routers.Get(s.scope.Project(), s.scope.Region(), getRouterName(s.scope.NetworkName())).Do()
```

Comment:
shall we only delete the router when it is auto created by us?

Reply:
This flow is executed only when the network has **owned** tag. https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/f321e5d79c59b5122af808cd3d62d9e4b31f4749/cloud/services/compute/network.go#L86